src/ray/raylet/node_manager.cc

```diff
@@ -261,6 +261,35 @@ void NodeManager::ProcessNewClient(std::shared_ptr<LocalClientConnection> client
   client->ProcessMessages();
 }
 
+void NodeManager::DispatchTasks() {
+  // Work with a copy of scheduled tasks.
+  auto scheduled_tasks = local_queues_.GetScheduledTasks();
+  // Return if there are no tasks to schedule.
+  if (scheduled_tasks.empty()) {
+    return;
+  }
+  const ClientID &my_client_id = gcs_client_->client_table().GetLocalClientId();
+
+  for (const auto &task : scheduled_tasks) {
+    const auto &local_resources =
+        cluster_resource_map_[my_client_id].GetAvailableResources();
+    if (local_resources.IsEmpty()) {
+      // Early return if there are no resources available.
+      return;
```

Comment:
Hmm, I just realized there's an issue here where a task that requires zero resources (e.g., an actor task) might not get scheduled. Was that why you had this check outside of the loop at first? Although even having it outside of the loop wouldn't fix this issue...

Not sure what the best approach here is. Should we just always loop through all the tasks for now?

Reply:
honestly, I think zero-resource tasks are just wrong... How can you have an active thread of execution running without having it acquire at least some CPU resource?
Yes, this is why I was asking about actor tasks. Yes, I think we have no choice but to drop the early termination check.